From: Ondrej Holy <oholy@redhat.com>
Date: Fri, 21 Sep 2018 16:12:51 +0200
Subject: glocalfile: Use MAXSYMLINKS when following symlinks

Currently, readlink() is used only 12 times when expanding symlinks.
However, kernel uses 40 for this purpose and it is defined as MAXSYMLINKS.
Use that constant if available, or 40. See:
https://github.com/torvalds/linux/include/linux/namei.h.

Applied-upstream: 2.58.2, commit:2daebb93fb1a2b510eaea3393b12d99556389849
---
 gio/glocalfile.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/gio/glocalfile.c b/gio/glocalfile.c
index a547bcd..0647559 100644
--- a/gio/glocalfile.c
+++ b/gio/glocalfile.c
@@ -1610,7 +1610,15 @@ expand_symlinks (const char *path,
         }
 
       num_recursions++;
-      if (num_recursions > 12)
+
+#ifdef MAXSYMLINKS
+      if (num_recursions > MAXSYMLINKS)
+#else
+      /* 40 is used in kernel sources currently:
+       * https://github.com/torvalds/linux/include/linux/namei.h
+       */
+      if (num_recursions > 40)
+#endif
         {
           g_free (target);
           return NULL;
